src/net_processing.cpp

```diff
@@ -1499,7 +1499,7 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
     std::deque<CInv>::iterator it = pfrom->vRecvGetData.begin();
     std::vector<CInv> vNotFound;
     const CNetMsgMaker msgMaker(pfrom->GetSendVersion());
-    {
+    if (pfrom->m_tx_relay != nullptr) {
```

Comment:
Would be nice to add a comment here to explain that this will effectively treat MSG_TX and MSG_WITNESS_TX as "unknown" types, thus blocking the queue, thus starving this peer completely and making it a zombie.

Reply:
Addressed in latest commit.